### DISCUSSION TITLE
Latest Observation Errors

### DISCUSSION THREAD

**Discussion by colonieweather** [weather-gov/api#233] [Q&A]
Over the last week or so, I've notice in the latest observation JSON file, that the text description and icon fields are frequently blank or null.  The last hour precipitation element is also missing.

[json_error.txt](https://github.com/weather-gov/api/files/5944081/json_error.txt)

**Comment by StephenClouse** [comment]
We are being told this is an upstream issue with MADIS and they are working on a fix. We do not have an ETA at present.

If you need a temporary fix I would recommend disregarding these five-minute observations and using the full /observations feed to retrieve the hourly METAR instead of /latest.

**Comment by colonieweather** [comment]
Thanks Stephen!

So you are saying to use for example this - https://api.weather.gov/stations/KALB/observations/
instead of this - https://api.weather.gov/stations/KALB/observations/latest 

Looks like the same errors are appearing in both.

**Comment by drewcoll** [comment]
Having the same issue on our website - but realized it was a common browser plugin blocking urls.

**Comment by scadergit** [comment]
@colonieweather as @StephenClouse noted, you have to filter for the hourly requests, that have a cloudLayer value, which is what derrvies the icon (it will also have a valid rawMessage). For KALB it's every observation at XX:51.

**Comment by JJKraw** [comment]
Question: Is this functioning as intended or should there be an endpoint that returns the latest full METAR?

Thanks!

**Comment by scadergit** [comment]
@JJKraw the /latest would typically function that way aside from the upstream MADIS issue.
